### PR TITLE
Feat/recommendationFilter

### PR DESCRIPTION
**Pull request by niklas1531**

### CODE REVIEW

File: src/data/apischema.py [hunk 1]
```diff
@@ -42,9 +44,14 @@ class GetRecommendationFilter(BaseModel):
     task_id: Optional[int] = None
     date: Optional[str] = None
     location: Optional[str] = None
-    severity: Optional[str] = None
```
Comment (ishworgiri1999) [inline src/data/apischema.py:45]: why not make it a  dict with min, max attribute ?

File: src/routes/recommendations.py [hunk 2]
```diff
@@ -67,14 +67,26 @@ def recommendations(
...
 
     total_count = finding_repository.get_findings_count_by_task_id(task.id)
+    
+    recommendations = [
+    db_finding_to_response_item(find)
+    for find in findings
+    if severity_range is None or (
+        (response_item := db_finding_to_response_item(find)).severity is not None and
+        len(severity_range) == 2 and
+        severity_range[0] <= response_item.severity <= severity_range[1]
```
Comment (ishworgiri1999) [inline src/routes/recommendations.py:77]: It probably makes sense to do it on the database level than the application level. We have severity stored in findings.

File: src/data/apischema.py (hunk 1 above)
Comment (niklas1531) [reply src/data/apischema.py:45]: true. changed it

File: src/repository/finding.py [hunk 3]
```diff
@@ -33,6 +34,25 @@ def get_findings_by_task_id(
...
+    def get_findings_by_task_id_and_severity(
+        self, task_id: int, severity: SeverityFilter, pagination: PaginationInput
+    ) -> list[db_models.Finding]:
+        findings = (
+            self.session.query(db_models.Finding)
+            .join(db_models.RecommendationTask)
+            .where(
+                db_models.RecommendationTask.status == db_models.TaskStatus.COMPLETED,
+                (db_models.RecommendationTask.id == task_id),
+                db_models.Finding.severity >= severity.minValue,
```
Comment (niklas1531) [inline src/repository/finding.py:47]: @ishworgiri1999 Do you understand why the comparison doesn't work? When I do this, my findings are always empty😟

Comment (ishworgiri1999) [reply src/repository/finding.py:47]: @niklas1531  I never saved severity, and priority to db. hopefully, it should work now.

**Review by ishworgiri1999 — APPROVED** [review]

File: src/repository/finding.py (hunk 3 above)
Comment (niklas1531) [reply src/repository/finding.py:47]: Unfortunately not

**Pull request merged**